src/libsync/discovery.cpp

```diff
@@ -34,6 +37,61 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcDisco, "sync.discovery", QtInfoMsg)
 
+bool ProcessDirectoryJob::checkForInvalidFileName(const SyncFileItemPtr &item)
+{
+    const QString folderPath = _discoveryData->_localDir;
+
+    const auto originalFilePathSyncFolder = item->_file;
+    auto slashPosition = originalFilePathSyncFolder.lastIndexOf('/');
+    const auto originalFileParentPathSyncFolder = slashPosition >= 0 ? originalFilePathSyncFolder.left(slashPosition) : QString();
+    const auto newFileName = originalFilePathSyncFolder.mid(slashPosition + 1).trimmed();
+    const QString originalFilePathAbsolute = folderPath + item->_file;
+    slashPosition = originalFilePathAbsolute.lastIndexOf('/');
+    const auto orignalFileParentPathAbsolute = originalFilePathAbsolute.left(slashPosition);
+    const QString newFilePathAbsolute = orignalFileParentPathAbsolute + QStringLiteral("/") + newFileName;
+
+    if (newFilePathAbsolute != originalFilePathAbsolute) {
```

Comment:
can you please add a new test that makes this condition true ?

Reply:
done :)